### PR TITLE
Alternate consumption function

### PR DESCRIPTION
**Pull request by aphearin**
**Do not merge:** This PR brings in an alternate version of the gas consumption kernel **_gas_conversion_kern**. The current implementation has an awkward way of defining a derivative in terms of a finite timestep and then normalizing with a clip. This alternate version first defines a _cumulative_ triweight function and then differentiates it with `jax.grad`. The idea is that the cumulative triweight automatically handles the unit normalization, and then using `jax.grad` guarantees exact consistency with the derivative and thereby alleviates the need to divide manually by dt. Besides being a bit more elegant, the current consumption function only works by dividing by the _mean_ value of some input dt array, which is not strictly correct for non-linearly-spaced arrays.

Note that this kernel is not quite the same, as shown in this figure, and so if we decide to implement this, all our fits will need to be rerun.
new_consumption

### CODE REVIEW

**Comment by aphearin** [comment]
For convenience [here](https://github.com/ArgonneCPAC/diffstar/blob/4d15a5b2fd2faa86311c543a151fee73a14bd7f1/diffstar/stars.py#L510) is the line where the normalization is computed by hand, and [here](https://github.com/ArgonneCPAC/diffstar/blob/4d15a5b2fd2faa86311c543a151fee73a14bd7f1/diffstar/stars.py#L437) is the line where we divide by the _mean_ value of the input `dtarr` array.

**Comment by alexalar** [comment]
Thanks a lot @aphearin ! Regarding the line where you say that we divide by the mean value of the input dtarr, I think I disagree. What [line 437](https://github.com/ArgonneCPAC/diffstar/blob/4d15a5b2fd2faa86311c543a151fee73a14bd7f1/diffstar/stars.py#L437) is doing is to create a matrix of ones when τ > 5*avg(dt) (and a matrix of zeros otherwise), and the avg(dt) is not being divided anywhere, and I don't think it's being used elsewhere.

Then the consumption matrix is the identity matrix where τ < 5\*avg(dt) and the calculated consumption matrix when τ < 5\*avg(dt) ([line 439](https://github.com/ArgonneCPAC/diffstar/blob/4d15a5b2fd2faa86311c543a151fee73a14bd7f1/diffstar/stars.py#L439)). This is done to avoid numerical issues that I had found in earlier development.

**Comment by aphearin** [comment]
Ahh, ok, yes, I see that now. Maybe this is not an issue then. Like I said originally, it may not be worth implementing this change. It is possible that the only reason I find my alternate grad-based implementation clearer is because I'm the one who wrote it.

**Comment by alexalar** [comment]
Note what we need for the gas consumption matrix is the integral of the gas consumption function over the finite time-step between snapshots. Also, note that the gas consumption function is just the triweight function, and then to calculate its integral we use the CDF of the triweight. We have both implemented analytically here. For the use of the CDF triweight to calculate the exact integral of the consumption function over the finite time-step for the consumption matrix, see the use of `tw_bin_jax_kern` in [line 516](https://github.com/ArgonneCPAC/diffstar/blob/4d15a5b2fd2faa86311c543a151fee73a14bd7f1/diffstar/stars.py#L516) which just calculates the difference of the triweight CDF `tw_cuml_jax_kern`. The normalization is calculated also with the same `tw_bin_jax_kern` CDF difference in [line 510 ](https://github.com/ArgonneCPAC/diffstar/blob/4d15a5b2fd2faa86311c543a151fee73a14bd7f1/diffstar/stars.py#L510). 

I think you are reading this finite integral of the CDF triweight as a way to calculate a derivative at a particular t_obs.

**Comment by aphearin** [comment]
Yes, I completely agree. I now realize that your implementation does *not* compute the normalization via an integral. I think the only differences between your function and mine are purely cosmetic. For me this PR was worth doing if only to improve my understanding of how the gas consumption function kernels work in detail, and so we may wish to just close this PR without merging if you decide that these cosmetic differences are not worth the trouble of rerunning our fits. I leave this entirely up to you.

**Comment by aphearin** [comment]
Alex convinced me that this function is slightly less accurate than the current one, and so there's really not a good reason to make this change.
